pkg/migration/migrations/separated_intents.go

```diff
@@ -347,10 +373,27 @@ func (m *migrateLockTablePool) runStatusLogger(ctx context.Context) {
 			}
 
 			finished := atomic.LoadUint64(&m.finished)
-			log.Infof(ctx, "%d ranges have completed lock table migration", finished)
+			total := m.total // TODO(dt): count in background and load via atomic.
+			lowWaterMark := m.lowWaterMark()
+			log.Infof(ctx, "%d of %d ranges, up to %s, have completed lock table migration", finished, total, lowWaterMark)
 			if ranges.Len() > 0 {
 				log.Infof(ctx, "currently migrating lock table on ranges %s", ranges.String())
 			}
+			if m.job != nil && timeutil.Since(lastProgress) > 30*time.Second {
+				lastProgress = timeutil.Now()
+				if err := m.job.FractionProgressed(ctx, nil, func(ctx context.Context, details jobspb.ProgressDetails) float32 {
+					prog := details.(*jobspb.Progress_Migration).Migration
+					prog.Watermark = lowWaterMark
+					if total > 0 && finished <= total {
+						return float32(finished) / float32(total)
```

Comment:
Btw we spin off 2 requests per range, so `finished` will actually end at `2*total`. Might have to update this fraction as well as the log line.

Reply:
this made me realize that I actually needed to change it up a little so we only consider a range "done" if we do both. Added a new commit that switches the channel to have ranges on it as the unit of work, then the workers do both the local and global to "complete" that unit, rather than doing them separately. That makes it easier to track what is done and what isn't, and makes the number just num ranges now, not something between numRanges and numRanges*2 based on which do/don't have global keys.